Vm boot options (#607)
vmware/go-vcloud-director: .changes/v2.22.0/607-features.md

> * Added `firmware` field to `VmSpecSection`, support specifying Boot Options of a VM [GH-607]

Author: Updated, also created 607-deprecations.md as suggested

> * Added `firmware` field to `VmSpecSection` type, `BootOptions` to `Vm` type,
> New functions `*Vdc.GetHardwareVersion`, `*Vdc.GetHighestHardwareVersion`, 
> `*Vdc.FindOsFromId`, `*VM.UpdateBootOptions`, `*VM.UpdateBootOptionsAsync` [GH-607]

Reviewer: ```suggestion
* Added `firmware` field to `VmSpecSection` type and `BootOptions` to `Vm` type [GH-607]
* Added `Vdc` methods `GetHardwareVersion`, `GetHighestHardwareVersion`, and `FindOsFromId` [GH-607]
* Added `VM` methods `UpdateBootOptions` and `UpdateBootOptionsAsync` [GH-607]
```

Author: Changed, thanks.

> * API calls for `AddRawVM`, `CreateStandaloneVmAsync`, `*VM.Refresh()`, 
> `*VM.UpdateVmSpecSectionAsync`, `addEmptyVmAsyncV10`, `getVMByHrefV10` 

Reviewer: ```suggestion
* API calls for `AddRawVM`, `CreateStandaloneVmAsync`, `VM.Refresh`, 
`VM.UpdateVmSpecSectionAsync`, `addEmptyVmAsyncV10`, `getVMByHrefV10` 
```

Author: Changed

> * Added `firmware` field to `VmSpecSection` type and `BootOptions` to `Vm` type [GH-607]

Reviewer 2: It should be capital "Firmware", doesn't it?

Author: Yes, updated.

* Added `Vdc` methods `GetHardwareVersion`, `GetHighestHardwareVersion`, 
`FindOsFromId` [GH-607] 
* Added `VM` methods `UpdateBootOptions`, `UpdateBootOptionsAsync` [GH-607]
* API calls for `AddRawVM`, `CreateStandaloneVmAsync`, `VM.Refresh`, 
`VM.UpdateVmSpecSectionAsync`, `addEmptyVmAsyncV10`, `getVMByHrefV10` 

> and `UpdateBootOptionsAsync` get elevated to API version `37.1` if available, for `firmware` and `BootOptions` support [GH-607]

Reviewer 2: ```suggestion
and `UpdateBootOptionsAsync` get elevated to API version `37.1` if available, for `Firmware` and `BootOptions` support [GH-607]
```

Author: Updated to `VmSpecSection.Firmware`
